### PR TITLE
chore: Convert Screens to Functional Components

### PR DESCRIPTION
**Pull request by cmejet**
Converts Login, Registration and Intro screens to functional components to role model the current best practices.

### CODE REVIEW

File: template/.vscode/settings.json [hunk 1]
```diff
@@ -26,5 +26,8 @@
   },
   "[typescriptreact]": {
     "editor.formatOnSave": false
```
Comment (cball) [inline template/.vscode/settings.json]: do we still want to set this to false?

File: template/.vscode/settings.json [hunk 2]
```diff
@@ -26,5 +26,8 @@
...
+  },
+  "editor.codeActionsOnSave": {
+    "source.fixAll.eslint": true
```
Comment (cball) [inline template/.vscode/settings.json]: 👍

**Comment by cball** [comment]
@cmejet related, have we removed the prompt in the generators?

File: template/.vscode/settings.json (hunk 1 above)
Comment (mcavaliere) [reply template/.vscode/settings.json]: I'd vote for true, this is super handy

File: template/src/utils/trimText/trimText.test.ts [hunk 3]
```diff
@@ -1,14 +1,14 @@
 import { trimText } from './trimText';
 
 describe('trimText', () => {
-  test('should trim a long string to the alloted length', () => {
+  it('should trim a long string to the alloted length', () => {
```
Comment (mcavaliere) [inline template/src/utils/trimText/trimText.test.ts]: ❤️

File: template/.vscode/settings.json (hunk 1 above)
Comment (cmejet) [reply template/.vscode/settings.json]: Dang it that is what it should have been. It's updating to the new prop and I copy pasta wrong.

**Comment by cmejet** [comment]
@all-contributors please add @mcavaliere for code and review

File: template/.vscode/settings.json (hunk 1 above)
Comment (cmejet) [reply template/.vscode/settings.json]: I will get this in the next PR I put up, I know there will be more to update.

**Pull request merged**